### PR TITLE
[Bugfix] Persist the maps in mv

### PR DESCRIPTION
**Pull request by blackstar-baba**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

**Comment by Astralidea** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 1]
```diff
@@ -2280,6 +2272,52 @@ public void replayCreateMaterializedView(String dbName, MaterializedView materia
         }
     }
 
+    public void replayAddMvPartitionNameRef(MaterializedViewPartitionNameRef ref) {
+        Database db = this.idToDb.get(ref.getDbId());
+        MaterializedView mv = ((MaterializedView) db.getTable(ref.getMvId()));
+        db.writeLock();
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:2278]: no need to lock for replay.

File: fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedViewPartitionNameRef.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import com.google.gson.annotations.SerializedName;
+import com.starrocks.common.io.Text;
+import com.starrocks.common.io.Writable;
+import com.starrocks.persist.gson.GsonUtils;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+
+public class MaterializedViewPartitionNameRef implements Writable {
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedViewPartitionNameRef.java:13]: what is the difference between this class and MaterializedViewPartitionRef?

**Review by ABingHuang — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java (hunk 1 above)
Comment (blackstar-baba) [reply fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:2278]: I also don't think it's needed if the journals are executed sequentially, but some replay functions added it.  e.g.
```
 public void replayAddPartition(PartitionPersistInfo info) throws DdlException {
        Database db = this.getDb(info.getDbId());
        db.writeLock();
        try {
            OlapTable olapTable = (OlapTable) db.getTable(info.getTableId());
            Partition partition = info.getPartition();

            PartitionInfo partitionInfo = olapTable.getPartitionInfo();
            if (info.isTempPartition()) {
                olapTable.addTempPartition(partition);
            } else {
                olapTable.addPartition(partition);
            }
```

File: fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedViewPartitionNameRef.java (hunk 2 above)
Comment (blackstar-baba) [reply fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedViewPartitionNameRef.java:13]: this used for persist partition name map, `MaterializedViewPartitionRef` used for  persist version map

**Comment by Astralidea** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java (hunk 1 above)
Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:2278]: it is not necessary. remove it

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 3]
```diff
@@ -2280,6 +2272,52 @@ public void replayCreateMaterializedView(String dbName, MaterializedView materia
...
+            mv.addPartitionNameRef(ref.getTablePartitionName(), ref.getMvPartitionName(), true);
+        } finally {
+            db.writeUnlock();
+        }
+    }
+
+    public void replayRemoveMvPartitionNameRef(MaterializedViewPartitionNameRef ref) {
+        Database db = this.idToDb.get(ref.getDbId());
+        MaterializedView mv = ((MaterializedView) db.getTable(ref.getMvId()));
+        db.writeLock();
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:2289]: no need to lock in replay

File: fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedViewPartitionRef.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import com.google.gson.annotations.SerializedName;
+import com.starrocks.common.io.Text;
+import com.starrocks.common.io.Writable;
+import com.starrocks.persist.gson.GsonUtils;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+
+public class MaterializedViewPartitionRef implements Writable {
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedViewPartitionRef.java:13]: is MaterializedViewPartitionVersionInfo or something a better name?

**Comment by blackstar-baba** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 80 / 147 (54.42%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|0|26|00.00%|[2276, 2277, 2278, 2280, 2282, 2284, 2287, 2288, 2289, 2291, 2293, 2295, 2298, 2299, 2300, 2302, 2303, 2304, 2306, 2308, 2311, 2312, 2313, 2315, 2317, 2319]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|8|00.00%|[2309, 2310, 2313, 2314, 2317, 2318, 2321, 2322]|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|0|6|00.00%|[202, 203, 204, 207, 208, 209]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|8|20|40.00%|[191, 192, 193, 196, 197, 198, 201, 202, 203, 206, 207, 208]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|20|26|76.92%|[396, 397, 405, 406, 407, 408]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedViewPartitionVersionInfo.java|28|34|82.35%|[74, 78, 79, 90, 91, 102]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedViewPartitionNameRefInfo.java|23|26|88.46%|[68, 69, 80]|
|:large_blue_circle:|com/starrocks/scheduler/MvTaskRunProcessor.java|1|1|100.00%|[]|

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Review by Astralidea — DISMISSED** [review]

**Review by ABingHuang — DISMISSED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 83 / 153 (54.25%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|0|29|00.00%|[2488, 2489, 2490, 2492, 2494, 2496, 2499, 2500, 2501, 2503, 2504, 2505, 2506, 2509, 2511, 2514, 2515, 2516, 2518, 2519, 2520, 2522, 2524, 2527, 2528, 2529, 2531, 2533, 2535]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|8|00.00%|[2278, 2279, 2282, 2283, 2286, 2287, 2290, 2291]|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|0|6|00.00%|[203, 204, 205, 208, 209, 210]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|8|20|40.00%|[199, 200, 201, 204, 205, 206, 209, 210, 211, 214, 215, 216]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|24|30|80.00%|[404, 405, 413, 414, 415, 416]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedViewPartitionVersionInfo.java|28|34|82.35%|[74, 78, 79, 90, 91, 102]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedViewPartitionNameRefInfo.java|23|26|88.46%|[68, 69, 80]|

**Review by ABingHuang — APPROVED** [review]
LGTM

**Review by chaoyli — APPROVED** [review]

**Pull request merged**